### ISSUE TITLE
We're having a year XYZ problem ;-)

### ISSUE THREAD

**Issue by DDvO** [openssl/openssl#13765]
After New Year, the CI test `check_update` fails as quoted below.
I've just provided a workaround for 2021: #13764 which does not solve the actual issue.

```
Run git diff --exit-code
diff --git a/crypto/asn1/charmap.h b/crypto/asn1/charmap.h
index e855b159..ac1eb076 100644
--- a/crypto/asn1/charmap.h
+++ b/crypto/asn1/charmap.h
@@ -2,7 +2,7 @@
  * WARNING: do not edit!
  * Generated by crypto/asn1/charmap.pl
  *
- * Copyright 2000-2020 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2000-2021 The OpenSSL Project Authors. All Rights Reserved.
  *
  * Licensed under the Apache License 2.0 (the "License").  You may not use
  * this file except in compliance with the License.  You can obtain a copy
diff --git a/crypto/bn/bn_prime.h b/crypto/bn/bn_prime.h
index ef16bb43..8a859ac0 100644
--- a/crypto/bn/bn_prime.h
+++ b/crypto/bn/bn_prime.h
@@ -2,7 +2,7 @@
  * WARNING: do not edit!
  * Generated by crypto/bn/bn_prime.pl
  *
- * Copyright 1998-2020 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 1998-2021 The OpenSSL Project Authors. All Rights Reserved.
  *
  * Licensed under the Apache License 2.0 (the "License").  You may not use
  * this file except in compliance with the License.  You can obtain a copy
diff --git a/crypto/conf/conf_def.h b/crypto/conf/conf_def.h
index 3fdb6a9b..1f66a58e 100644
--- a/crypto/conf/conf_def.h
+++ b/crypto/conf/conf_def.h
@@ -2,7 +2,7 @@
  * WARNING: do not edit!
  * Generated by crypto/conf/keysets.pl
  *
- * Copyright 1995-2020 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 1995-2021 The OpenSSL Project Authors. All Rights Reserved.
  * Licensed under the Apache License 2.0 (the "License").  You may not use
  * this file except in compliance with the License.  You can obtain a copy
  * in the file LICENSE in the source distribution or at
diff --git a/crypto/objects/obj_dat.h b/crypto/objects/obj_dat.h
index 440fd1d6..1b852e6d 100644
--- a/crypto/objects/obj_dat.h
+++ b/crypto/objects/obj_dat.h
@@ -2,7 +2,7 @@
  * WARNING: do not edit!
  * Generated by crypto/objects/obj_dat.pl
  *
- * Copyright 1995-2020 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 1995-2021 The OpenSSL Project Authors. All Rights Reserved.
  * Licensed under the Apache License 2.0 (the "License").  You may not use
  * this file except in compliance with the License.  You can obtain a copy
  * in the file LICENSE in the source distribution or at
diff --git a/crypto/objects/obj_xref.h b/crypto/objects/obj_xref.h
index ba290cc6..0f8a0565 100644
--- a/crypto/objects/obj_xref.h
+++ b/crypto/objects/obj_xref.h
@@ -2,7 +2,7 @@
  * WARNING: do not edit!
  * Generated by objxref.pl
  *
- * Copyright 1998-2020 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 1998-2021 The OpenSSL Project Authors. All Rights Reserved.
  *
  * Licensed under the Apache License 2.0 (the "License").  You may not use
  * this file except in compliance with the License.  You can obtain a copy
diff --git a/include/openssl/obj_mac.h b/include/openssl/obj_mac.h
index 5af00249..89b44903 100644
--- a/include/openssl/obj_mac.h
+++ b/include/openssl/obj_mac.h
@@ -2,7 +2,7 @@
  * WARNING: do not edit!
  * Generated by crypto/objects/objects.pl
  *
- * Copyright 2000-2020 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2000-2021 The OpenSSL Project Authors. All Rights Reserved.
  * Licensed under the Apache License 2.0 (the "License").  You may not use
  * this file except in compliance with the License.  You can obtain a copy
  * in the file LICENSE in the source distribution or at
Error: Process completed with exit code 1.
```

**Comment by mattcaswell** [comment]
Not the first time:
#10738, #7972

Although apparently "make update" was sufficient on those occasions.

**Comment by richsalz** [comment]
It seems that the scripts that build the generated files have bugs.  The copyright year should only be updated if either the *script itself* changes, or the input source file changes.  Looking at crypto/asn1/charmap.pl it uses the current date:
```
# Year the file was generated.
my $YEAR = [localtime()]->[5] + 1900;
```

Looking at 339638b586e77c189e8294f931b99b03d1f08dbd it looks like i did this on purpose:
```
Author: Rich Salz <rsalz@akamai.com>
Date:   Tue Jan 7 15:50:20 2020 -0500

    Make generated copyright year be "now"
    
    Always use the current year in generating output files, rather than
    trying to base is on the modtime of the script or input, as that can
    vary depending on the ability of the local OS to keep those accurate.
    
    Fixes #10744
```

Perhaps the project should now require git clients to require accurate timestamp?  Or enhance the scripts/makefile to let you explicitly pass in a year for those that don't do so?

**Comment by mattcaswell** [comment]
Or use the year from the last commit that changed the script or input?

**Comment by richsalz** [comment]
> Or use the year from the last commit that changed the script or input?

That's a good idea, but it should fallback cleanly for those who just download the tarballs, not from git.

I can work on this if nobody else is itching to do so.

**Comment by DDvO** [comment]
This bug - predictably - has hit us again.

**check_update** ran in another instance of the New Year Problem:
```
Run git diff --exit-code
diff --git a/crypto/asn1/charmap.h b/crypto/asn1/charmap.h
index ac1eb076..95928ca6 100644
--- a/crypto/asn1/charmap.h
+++ b/crypto/asn1/charmap.h
@@ -2,7 +2,7 @@
  * WARNING: do not edit!
  * Generated by crypto/asn1/charmap.pl
  *
- * Copyright 2000-2021 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2000-2022 The OpenSSL Project Authors. All Rights Reserved.
  *
  * Licensed under the Apache License 2.0 (the "License").  You may not use
  * this file except in compliance with the License.  You can obtain a copy
```
etc.

Current workaround is in #17398.

**Comment by DDvO** [comment]
I wonder why  @openssl-machine had closed this in c37b947 on Mar 31, 2021.

I just added **milestone 3.1.0** hoping that this helps as reminder to fix this nuisance before it hits us again.
